test/jest/__tests__/utils/utils.DependencyUtils.ts.spec.ts

```diff
@@ -0,0 +1,280 @@
+                "author-mod-2.0.0",
+                "author-modWithDependency-1.0.0",
+                "author-modWithTheSameDependency-1.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod with a deep dependency", () => {
+        it("should return the mod and its dependency's last version and its dependency's dependency's latest version", async () => {
+            const combos = [modWithDeepDependency];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modWithDeepDependency-1.0.0",
+                "author-modWithDependency-2.0.0",
+                "author-mod-2.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod with a loop dependency", () => {
+        it("should return the mod's specific version and its dependency's last version", async () => {
+            const combos = [modWithLoopDependencyA];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modWithLoopDependencyA-1.0.0",
+                "author-modWithLoopDependencyB-2.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A mod with two dependencies that have the same dependency", () => {
+        it("should return the mod's specific version and both dependecies' last version and the deep dependant's last version", async () => {
+            const combos = [modWithTwoDependeciesThatHaveTheSameDependency];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modWithTwoDependeciesThatHaveTheSameDependency-1.0.0",
+                "author-modWithDependency-2.0.0",
+                "author-modWithTheSameDependency-2.0.0",
+                "author-mod-2.0.0"
+            ]);
+        });
+    });
+
+    
+
+    describe("Single mod with a dependency to a mod that's already installed", () => {
+        it("should return the mod and its dependency's latest version", async () => {
+            const combos = [modWithDependency];
+
+            await getFullDependencyList(combos, game, [new ManifestV2().fromThunderstoreCombo(mod)]);            
+
+            expectToContainOnly(combos, [
+                "author-modWithDependency-1.0.0"
+            ]);
+        });
+    });
+
+
+
+    describe("A modpack with a dependency", () => {
+        it("should return the modpack mod and the specific version of the dependency", async () => {
+            const combos = [modPack];
+
+            await getFullDependencyList(combos, game, []);
+
+            expectToContainOnly(combos, [
+                "author-modPack-1.0.0",
+                "author-mod-1.0.0"
+            ]);
+        });
+    });
+
+
+    describe("A modPack and a mod with the same dependency", () => {
+        it("should return the modPack's specific version, mod's specific version and the dependency's last version", async () => {
```

Comment:
Should it though? I'm not sure we have a spec that comments one way or another if the latest or specific version should be returned in a case like this? I might also be outside the scope of this PR.

Reply:
This is now updated, it should now at least work in the same way it worked before.